tests/components/file/test_notify.py

```diff
@@ -1,91 +1,78 @@
-from homeassistant.setup import setup_component
+from homeassistant.setup import async_setup_component
 import homeassistant.util.dt as dt_util
 
 from tests.async_mock import call, mock_open, patch
-from tests.common import assert_setup_component, get_test_home_assistant
-
-
-class TestNotifyFile(unittest.TestCase):
-    """Test the file notify."""
-
-    def setUp(self):  # pylint: disable=invalid-name
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.addCleanup(self.tear_down_cleanup)
-
-    def tear_down_cleanup(self):
-        """Stop down everything that was started."""
-        self.hass.stop()
+from tests.common import assert_setup_component
 
-    def test_bad_config(self):
-        """Test set up the platform with bad/missing config."""
-        config = {notify.DOMAIN: {"name": "test", "platform": "file"}}
-        with assert_setup_component(0) as handle_config:
-            assert setup_component(self.hass, notify.DOMAIN, config)
-        assert not handle_config[notify.DOMAIN]
 
-    def _test_notify_file(self, timestamp):
-        """Test the notify file output."""
-        filename = "mock_file"
-        message = "one, two, testing, testing"
-        with assert_setup_component(1) as handle_config:
-            assert setup_component(
-                self.hass,
-                notify.DOMAIN,
-                {
-                    "notify": {
-                        "name": "test",
-                        "platform": "file",
-                        "filename": filename,
-                        "timestamp": timestamp,
-                    }
-                },
-            )
-        assert handle_config[notify.DOMAIN]
+async def test_bad_config(hass):
+    """Test set up the platform with bad/missing config."""
+    config = {notify.DOMAIN: {"name": "test", "platform": "file"}}
+    with assert_setup_component(0) as handle_config:
+        assert await async_setup_component(hass, notify.DOMAIN, config)
+    assert not handle_config[notify.DOMAIN]
 
-        m_open = mock_open()
-        with patch(
-            "homeassistant.components.file.notify.open", m_open, create=True
-        ), patch("homeassistant.components.file.notify.os.stat") as mock_st, patch(
-            "homeassistant.util.dt.utcnow", return_value=dt_util.utcnow()
-        ):
 
-            mock_st.return_value.st_size = 0
-            title = (
-                f"{ATTR_TITLE_DEFAULT} notifications "
-                f"(Log started: {dt_util.utcnow().isoformat()})\n{'-' * 80}\n"
-            )
+@pytest.mark.parametrize(
+    "timestamp",
+    [
+        False,
+        True,
+    ],
+)
+async def test_notify_file(hass, timestamp):
+    """Test the notify file output."""
+    filename = "mock_file"
+    message = "one, two, testing, testing"
+    with assert_setup_component(1) as handle_config:
+        assert await async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)